### PR TITLE
DSG-2119: transaction definition delete route

### PR DESCRIPTION
**Pull request by jgnuvalence**
## Description
Ticket: https://dol-ewf.atlassian.net/browse/DSG-2119

Add route to be able to delete a transaction definition.

## Types of changes

Patch
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation change (documentation only, will not require redeployment)
- [ ] Refactor or cleanup (functionality unchanged)
Minor
- [ ] New feature (non-breaking change which adds functionality)
Major
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

- [ ] I have confirmed all acceptance criteria defined by the story have been met

### CODE REVIEW

File: service/src/main/java/io/nuvalence/workmanager/service/controllers/AdminApiDelegateImpl.java [hunk 1]
```diff
@@ -796,16 +805,43 @@ public ResponseEntity<List<DashboardCountsModel>> getTransactionDefinitionSetCou
...
+
     private PagedWorkflowModel generatePagedWorkflowModel(Page<WorkflowModel> workflows) {
         PagedWorkflowModel model = new PagedWorkflowModel();
         model.items(workflows.toList());
         model.setPagingMetadata(pagingMetadataMapper.toPagingMetadata(workflows));
         return model;
     }
 
-    private boolean validateDefaultFormConfigKeyNullOrEmpty(
+    private FormConfiguration createDefaultFormConfiguration(
```
Comment (Ortega-Dan) [inline service/src/main/java/io/nuvalence/workmanager/service/controllers/AdminApiDelegateImpl.java:831]: very nit picky but I wonder if this should rather go in the service. Just to consider, not a big deal.

File: service/src/test/java/io/nuvalence/workmanager/service/audit/DynamicDataChangedAuditHandlerTest.java [hunk 2]
```diff
@@ -96,6 +98,7 @@ class DynamicDataChangedAuditHandlerTest {
     @Autowired private TransactionDefinitionService transactionDefinitionService;
     @Autowired private TransactionService transactionService;
     @Autowired private EntityMapper entityMapper;
+    @Autowired private FormConfigurationService formConfigurationService;
```
Comment (Ortega-Dan) [inline service/src/test/java/io/nuvalence/workmanager/service/audit/DynamicDataChangedAuditHandlerTest.java:101]: looks like this one is not really in use.

**Review by Ortega-Dan — APPROVED** [review]

File: service/src/test/java/io/nuvalence/workmanager/service/audit/DynamicDataChangedAuditHandlerTest.java (hunk 2 above)
Comment (jgnuvalence) [reply service/src/test/java/io/nuvalence/workmanager/service/audit/DynamicDataChangedAuditHandlerTest.java]: I just removed this, thank you

File: service/src/main/java/io/nuvalence/workmanager/service/controllers/AdminApiDelegateImpl.java (hunk 1 above)
Comment (jgnuvalence) [reply service/src/main/java/io/nuvalence/workmanager/service/controllers/AdminApiDelegateImpl.java]: I agree, I moved it to the service

**Review by Ortega-Dan — APPROVED** [review]

**Pull request merged**